README.md

```diff
@@ -398,6 +399,49 @@ const ui = (
 
 This callback will only be called if `isOpen` is `true`.
 
+### modifyStateChange
+
+> `function(state: object, stateToBeSet: object)` | optional
+
+**🚨 This is a really handy power feature 🚨**
+
+This function will be called each time `downshift` sets its internal state
+(or calls your `onStateChange` handler for control props). It allows you to
+modify the state change that will take place which can give you fine grain
+control over how the component interacts with user updates without having to
+use [Control Props](#control-props).
+
+* `state`: The full current state of downshift.
+* `stateToBeSet`: The state that is about to be set (including the `type` or
+  origin of the change which has references in `Downshift.stateChangeTypes`
+  as in [`onStateChange`](#onstatechange).
+
+This is a pure function. You _might_ think of it as a simple reducer!
+
+```jsx
+const ui = (
+  <Downshift modifyStateChange={modifyStateChange}>
+    {/* your callback */}
+  </Downshift>
+)
+
+function modifyStateChange(state, stateToBeSet) {
+  // this prevents the menu from being closed when the user
+  // selects an item with the keyboard
+  switch (stateToSet.type) {
+    case Downshift.stateChangeTypes.keyDownEnter:
+    case Downshift.stateChangeTypes.clickItem:
```

Comment:
As I was copying your test over for the example I realized that if the user selected an item with the mouse it would still close and then realized we didn't have a `type` for that situation, so I added one in this PR :+1:

Reply:
I know, I planned to add `{ type:  Downshift.stateChangeTypes.itemClick }` [here](https://github.com/paypal/downshift/blob/master/src/downshift.js#L738) after that PR 😅.